### ISSUE TITLE
Deep parsing of expressions

### ISSUE THREAD

**Issue by csandor** [geographika/mappyfile#66]
Great work! Thanks!

This is mostly question/feature request. 

I wonder if it would be possible to parse expressions to a deeper level? Like parsing the expression itself to a tree json structure. It seems to me that mappyfile is already doing some validation on the expressions. Comparison/Regex/List type expressions are relaitvely simple compared to Mapserver style expressions.

The rationale: I'm making a transformer from mapfile to mapbox json style specs[1] and would need to convert expressions to filters. I'm using exclusively "Mapserver" expressions [2] in my mapfiles.

Any idea, suggestion is welcome!
[1] https://www.mapbox.com/mapbox-gl-js/style-spec/
[2] https://mapserver.org/mapfile/expressions.html#mapserver-expressions

**Comment by geographika** [comment]
Hi @csandor - glad you like the project!

A mappyfile plugin to convert MapServer Mapfiles to Mapbox styling to allow the same MapServer styling  on the server and client for vector tiles was one of the (too many..) side-projects I had half started. 

I came to the conclusion it might make sense in the long-run to try and get full Mapfile to SLD compatibility, and then use other libraries to convert from SLD into other styling formats (Mapbox, QGIS, Mapnik etc.). This may make it easier to maintain assuming SLD remains the "lingua franca" of map styling as SLD to X would be a more common use case than a direct Mapfile to X. 

SLD includes not only styling but the OGC filter spec, which roughly corresponds to the MapServer expressions you linked to above. I had a rough prototype of converting these to OGC filters, so I've uploaded the script to https://github.com/geographika/mappyfile/blob/master/docs/examples/parsing.py - this will give you an idea of how to write a transformer for expressions. 

mappyfile does break down the MapServer expressions fully into tokens. The standard mappyfile transformer only really needs to print these out as a string to write back to a Mapfile, but a custom transformer can be used to change the expression into any required data structure e.g. XML or JSON. 

Let me know how you get on. I'll repost here if I get a chance to create a "mappyfile-expressions" plugin in the future.
